### PR TITLE
feat: extend the organizer analytics dashboard

### LINKED ISSUE #4383: Feature: Organizer Analytics Dashboard
## Goal
Provide organizers with a unified analytics page accessible from the organizer sidebar, so they can quickly understand how their events are performing across registrations, ticket sales, proposals, check-ins, email engagement, follower growth etc.

## Current Behavior
- Analytics in Eventyay are scattered across isolated, per-event locations.
- Ticket and order statistics live inside the per-event Statistics plugin and show orders-by-day, revenue-over-time, and orders-by-product using Morris.js charts.
- Talk and proposal statistics live inside the per-event Orga module and show proposal pipeline, state, type, and track breakdowns using ApexCharts.
- Admin-wide analytics exist only in the super-admin dashboard and are not visible to regular organizers.
- The organizer sidebar currently has no analytics entry.

## Expected Behavior
- Organizers can access a new "Analytics" section directly from their sidebar dashboard menu.
- The system presents a unified analytics view showing aggregated metrics across all permitted events belonging to the organizer.
- Charts load efficiently using a mixed library approach: Morris.js for line/area trends and ApexCharts for donuts/pie charts.
- Event permissions are strictly enforced so organizers only see aggregate metrics for events they actually have permission to view.

## Features
- Direct navigation entry in the organizer sidebar menu via the common navigation configuration.
- Scoped event permission check filtering events through the user's permitted event list.
- Ticket Sales & Revenue section showing orders over time, status breakdown, cumulative revenue and top events by sales.
- Proposals & Sessions section showing proposal pipeline states, submission timeline based on LogEntry and events with pending proposals.
- Check-In Summary section listing check-in rates per event using vertical bars and displaying entry check-ins over time.
- Email Engagement section surfacing outgoing and failed email counts.
- Follower Growth section tracking total follower counts and weekly/monthly growth rates.

### PR DESCRIPTION
**Pull request by prayag78**
Closes #4383

## Summary
Extends the Organizer Analytics dashboard by implementing the remaining sections: Attendance Trends, Email Engagement and Follower Growth. It also consolidates date-parsing helpers and introduces comprehensive unit tests.

### New Dashboard Sections
- **Attendance Trends:** Added an attendance activity area chart tracking placed orders vs. attendee registrations over the last 30 days, complete with an interactive per-event dropdown selector.
- **Email Engagement:** Displays a per-event tabular report showing total outgoing and pending emails (`sent` vs. `queued`). Scoped strictly to users holding the `can_view_orders` permission.
- **Follower Growth:** Tracks total organizer follower counts and visualizes weekly and monthly follower growth trends. As follower counts are organizer-wide, this section is visible to all permitted organizer members.
Screenshot 2026-07-23 224553
Screenshot 2026-07-23 224634

### CODE REVIEW

File: app/eventyay/eventyay_common/views/organizer_analytics.py [hunk 1]
```diff
@@ -42,6 +152,164 @@ def _event_ids_for_orders(self):
...
+    def _event_ids_for_email_engagement(self):
+        if not hasattr(self, '_cached_event_ids_for_email_engagement'):
+            self._cached_event_ids_for_email_engagement = list(
+                self.request.user.get_events_with_permission('can_view_orders', request=self.request)
+                .filter(organizer=self.request.organizer)
+                .values_list('pk', flat=True)
+            )
+        return self._cached_event_ids_for_email_engagement
+
+    def _get_email_engagement_data(self):
```
Comment (Rachit7168) [inline app/eventyay/eventyay_common/views/organizer_analytics.py:129]: In _get_email_engagement_data() you are using .values('event_id', 'event__name') and in _get_attendance_event_options() you are using .values('pk', 'name').

Because Event.name is an I18nCharField, calling .values() on it causes Django to return the raw serialized JSON payload (e.g. {"en": "My Event", "de": "Mein Event"}). When you later do str(row['name']), it renders that raw JSON dictionary directly into the UI!

Recommendation: Avoid using .values() to fetch I18nCharField data.

For the attendance options, fetch the models with .only() so the localized string resolves correctly

**Review by Rachit7168 — CHANGES_REQUESTED** [review]
Thanks for working on this , Please address :

File: app/eventyay/eventyay_common/views/organizer_analytics.py (hunk 1 above)
Comment (prayag78) [reply app/eventyay/eventyay_common/views/organizer_analytics.py:129]: fixed

File: app/eventyay/eventyay_common/views/organizer_analytics.py [hunk 2]
```diff
@@ -42,6 +152,169 @@ def _event_ids_for_orders(self):
...
+            event_id: {
+                date_label: {'orders': 0, 'registrations': 0}
+                for date_label in date_labels
+            }
+            for event_id in event_ids
+        }
+
+        with scopes_disabled():
+            orders_by_day = list(
+                Order.objects.filter(
```
Comment (Rachit7168) [inline app/eventyay/eventyay_common/views/organizer_analytics.py:269]: These queries for the Attendance Trends chart do not filter by order status. As a result, cancelled and expired orders are incorrectly counted, which will severely inflate the attendance numbers. Please add `status__in=[Order.STATUS_PAID, Order.STATUS_PENDING]` to the `Order.objects.filter()` query (and similarly `order__status__in=[...]` for `OrderPosition.all.filter()`).

File: app/eventyay/eventyay_common/views/organizer_analytics.py [hunk 3]
```diff
@@ -42,6 +152,169 @@ def _event_ids_for_orders(self):
...
+            'email_engagement_rows': email_engagement_rows,
+            'has_email_engagement': bool(email_engagement_rows),
+        }
+
+    def _get_follower_growth_data(self):
+        current_timezone = timezone.get_current_timezone()
+        with scopes_disabled():
+            followers = OrganizerFollower.objects.filter(organizer=self.request.organizer)
+            weekly_rows = list(
+                followers.annotate(period=TruncWeek('created', tzinfo=current_timezone))
```
Comment (Rachit7168) [inline app/eventyay/eventyay_common/views/organizer_analytics.py:172]: These queries only return rows for periods that actually have new followers. Because this data is rendered as an Area chart, the frontend will interpolate missing weeks/months, making it visually appear as if followers were gained during empty periods. Please post-process `weekly_rows` and `monthly_rows` to fill in missing weeks/months with `{count: 0}` between the minimum and maximum dates (similar to how `_get_attendance_data` initializes a bucket for every day).

File: app/eventyay/eventyay_common/views/organizer_analytics.py [hunk 4]
```diff
@@ -42,6 +152,169 @@ def _event_ids_for_orders(self):
...
+                    datetime__gte=start,
+                    datetime__lt=end,
+                )
+                .annotate(day=TruncDate('datetime', tzinfo=current_timezone))
+                .values('event_id', 'day')
+                .annotate(count=Count('pk'))
+                .order_by('event_id', 'day')
+            )
+            registrations_by_day = list(
+                OrderPosition.all.filter(
```
Comment (Rachit7168) [inline app/eventyay/eventyay_common/views/organizer_analytics.py]: Using `OrderPosition.all.filter(...)` is confusing because `.all` looks like a queryset method instead of a manager. Since we need to filter out cancelled/expired orders anyway, please change this to the standard `OrderPosition.objects.filter(...)` and add the required `order__status__in=[Order.STATUS_PAID, Order.STATUS_PENDING]` filter.

**Review by Rachit7168 — CHANGES_REQUESTED** [review]
This PR does a great job structuring the new analytics components and setting up the queries for Follower Growth, Attendance, and Email Engagement. However, it introduces a critical correctness bug by failing to filter by `Order.status` in the Attendance Trends queries. This will drastically inflate attendance metrics by counting cancelled and expired orders. Furthermore, the Follower Growth queries fail to zero-fill empty weeks/months, causing the Area charts to display misleading visual interpolations. These data accuracy and presentation issues must be addressed before this code is production-ready.

**Review by Saksham-Sirohi — CHANGES_REQUESTED** [review]
Direction is fine, but `_project_attendance` is doing a lot of unnecessary `isinstance` / bool / Mapping guards. We already build those dicts ourselves — keep it simple.

Also watch query cost on this page (followers + email + attendance on top of what's already there). Make sure you're not pulling full follower rows into Python when an aggregate would do, and cover the new panels in tests (empty state, event filter, permission boundary).

File: app/eventyay/eventyay_common/views/organizer_analytics.py (hunk 3 above)
Comment (prayag78) [reply app/eventyay/eventyay_common/views/organizer_analytics.py:172]: fixed

File: app/eventyay/eventyay_common/views/organizer_analytics.py (hunk 2 above)
Comment (prayag78) [reply app/eventyay/eventyay_common/views/organizer_analytics.py:269]: fixed

**Review by Aqil-Ahmad — CHANGES_REQUESTED** [review]
1. the filter reloads the page, is it possible to avoid that?
2. I am not sure sure which emails are shown, but i think these are talk only emails. if yes, maybe you can clarify the heading. not sure about the queued emails as well
3. the followers chart shows wrong dates and 4 days are shown only. 

Image

**Review by Saksham-Sirohi — COMMENTED** [review]
Copilot flagged the following:

- Attendance/email/follower analytics use `scopes_disabled()` only for cross-event aggregation under organizer permission checks;
- No missing `scope(event=event)` issues spotted in the backend additions.

**Review by Saksham-Sirohi — APPROVED** [review]

**Review by ArnavBallinCode — APPROVED** [review]
